### PR TITLE
CANS-1036: added caregiver domain validation for empty Caregiver Name

### PR DESCRIPTION
**Pull request by LeonidMarushevskyi**
## JIRA

https://osi-cwds.atlassian.net/browse/CANS-1036

## Description

Added caregiver domain validation for empty Caregiver Name

## Tests
- [ ] I used TDD to develop the feature
- [ ] I have included unit tests
- [x] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters and static analyzers for the project (SonarQube, eslint, rubocop, etc).
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java [hunk 1]
```diff
@@ -159,6 +180,19 @@ private boolean areItemsValid(
...
+    return assessment
+        .getDomains()
+        .stream()
+        .filter(isCaregiverDomainWithEmptyNameFilter())
+        .collect(Collectors.toList());
+  }
+
+  private Predicate<DomainJson> isCaregiverDomainWithEmptyNameFilter() {
+    return domain ->
+        (isTrue(domain.getIsCaregiverDomain())) && (domain.getCaregiverName().isEmpty());
```
Comment (denys-davydov) [inline src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: Possible NullPointerException at `domain.getCaregiverName()`

**Comment by DmitriyRudenko** [comment]
Main problem is a multiple iterations over domains. It needs to be replaced by single loop or by drools usage

**Comment by LeonidMarushevskyi** [comment]
Let's accept this as a minor technical debt. I don't see this as a critical problem because the number of domains is small and it will not cause any performance impact. Also, I just added one validation and refactoring of existing validations is out of scope for this change.

**Comment by denys-davydov** [comment]
@DmitriyRudenko From a trade-off perspective of (1) how much execution time we save and how much it is critical to save microseconds on additional 13 elements array iteration vs. (2) clear code readability hence not increased cost of support, I would prefer in this particular place and situation to have a cleaner code, option 2.

**Pull request closed without merging**